src/execution_plan/optimizations/full_scan_to_label_scan.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2018-2020 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#pragma once
+
+#include "../ops/ops.h"
+#include "../execution_plan.h"
+
+/* The fullScantoLabelScan optimization searches for
```

Comment:
I think that if we would update our `QueryGraph` at an early stage on the execution-plan construction
such that we take into account `WHERE a:L` type of filters and update the appropriate `qgnode` we would get the same effect for "free".

A few catches:
1. I don't think we have our filter-tree broken down to its smallest components at the required stage.
2. What if the specified filter is of the form: `WHERE a:L0 AND a:L1`

Another reason to push in this direction is:
`MATCH (n:L {id:1})-[]->(z) WHERE z:L RETURN *`
In this case we would like to have the `L` label matrix as part of the conditional-traverse operation, once again we can get this for free if the `qgnode` `z` is labeled as `L`